packages/landing/src/map.ts

```diff
@@ -68,15 +86,55 @@ export class Basemaps {
         }
         const view = new View({ projection: projection.toEpsgString(), center: loc, zoom: location.zoom, resolutions });
 
-        this.map = new Map({
+        const source = this.getSource();
+        source.addEventListener('tileloadstart', this.trackTileLoad);
+        source.addEventListener('tileloadend', this.trackTileLoad);
+        const layer = new TileLayer({ source });
+        this.map = new OlMap({
             target: this.el,
             view,
-            layers: [this.getLayer()],
+            layers: [layer],
         });
 
         this.map.addEventListener('postrender', this.postRender);
     }
 
+    tileTrackTimer: unknown | null = null;
+    trackTileLoad = (evt: BaseEvent): void => {
+        if (!isTileLoadEvent(evt)) return;
+        const metricName = 'tile:' + evt.tile.getTileCoord().join('-');
+        if (evt.type == 'tileloadstart') {
+            this.tileTimer.set(metricName, Date.now());
+            return;
+        }
+
+        if (evt.type == 'tileloadend') {
+            const startTime = this.tileTimer.get(metricName);
+            if (startTime == null) return;
+            this.tileTimer.delete(metricName);
+            const duration = Date.now() - startTime;
+            this.tileLoadTimes.push(duration);
+            if (this.tileTrackTimer == null) {
+                this.tileTrackTimer = setTimeout(this.reportTileStats, TileReportTimeDurationMs);
+            }
+        }
+    };
+
+    reportTileStats = (): void => {
+        this.tileTrackTimer = null;
+        const tileLoadTimes = this.tileLoadTimes;
+        if (tileLoadTimes.length < TileMinReportCount) return;
+        this.tileLoadTimes = [];
```

Comment:
Invalid reference to `this` from within a function that is called without binding `this` to it.

Reply:
this is a `() => {}` defined so `this` is bound at constructor time.